src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransportFactory.php

```diff
@@ -29,17 +29,21 @@ public function create(Dsn $dsn): TransportInterface
 
         $transport = new EsmtpTransport($host, $port, $tls, $this->dispatcher, $this->logger);
 
-        if ('' !== $dsn->getOption('verify_peer') && !filter_var($dsn->getOption('verify_peer', true), \FILTER_VALIDATE_BOOL)) {
-            /** @var SocketStream $stream */
-            $stream = $transport->getStream();
-            $streamOptions = $stream->getStreamOptions();
+        /** @var SocketStream $stream */
+        $stream = $transport->getStream();
+        $streamOptions = $stream->getStreamOptions();
 
+        if ('' !== $dsn->getOption('verify_peer') && !filter_var($dsn->getOption('verify_peer', true), \FILTER_VALIDATE_BOOL)) {
             $streamOptions['ssl']['verify_peer'] = false;
             $streamOptions['ssl']['verify_peer_name'] = false;
```

Comment:
Maybe we should set the `peer_fingerprint` to `false` here as well?

Reply:
`verify_peer` verifies certificate content, while `peer_fingerprint` works independently (with or without `verify_peer`) and verifies the certificate by hash without looking into it.

My motivation of this patch was actually to use `peer_fingerprint` with `verify_peer=false`, to ensure server authenticity even when using self-signed certs.